_dev/apps/ui/src/store/modules/product-feed/actions.ts

```diff
@@ -50,6 +52,18 @@ export const createProductFeedApiPayload = (settings:any) => ({
   requestSynchronizationNow: settings.requestSynchronizationNow,
 });
 
+const filterEstimateCarriers = (data: CustomCarrier[]) => {
+  if (data[0].offer === OfferType.FLAT_SHIPPING_RATE || data[0].offer === OfferType.FREE_SHIPPING) {
+    delete data[0].freeShippingOverAmount;
+    if (data[0].offer === OfferType.FREE_SHIPPING) {
+      delete data[0].flatShippingRate;
+    }
+  } else if (data[0].offer === OfferType.FREE_SHIPPING_OVER_AMOUNT) {
+    delete data[0].flatShippingRate;
+  }
```

Comment:
This change will fix the first element of the array. It is possible to estimate several carriers in the product feed, for instance by having a carrier by country.

You should apply your fix to each line.